src/libbson/src/bson/bson.c

```diff
@@ -3249,7 +3333,9 @@ bson_array_as_json (const bson_t *bson, size_t *length)
       return NULL;
    }
 
-   bson_string_append (state.str, " ]");
+   if (!state.max_len_reached) {
```

Comment:
Similarly, should this check that there is room for the last two characters? E.g. `state.str->len + 2 <= state.max_len`?

Reply:
`bson_array_as_json` is existing API, so we can't change the behaviour here without a BC break, hence always using `BSON_MAX_LEN_UNLIMITED`. This method is not used at all by libmongoc itself, so I don't think there's a use-case for adding the limit there. If there is, there should be a `bson_array_as_json_with_opts` that takes a `bson_json_opts_t` to pass the length, as was done with the other `bson_as_*` methods.

Reply:
@alcaeus that makes a lot of sense!  Why not leave this code the way it was before, then?  It feels a bit unnecessary to me to check if we've reached max length if we don't ever have a length limit.  Do you feel there's a benefit to leaving this logic in?

Reply:
Sorry for the late reply, I lost track of this comment thread. We can remove the logic from this method since it will never deal with max length and can add this back in if we do decide that we want to support this for `bson_array_as_json` eventually.

Reply:
Done.